docs/process_pages.py

```diff
@@ -215,6 +217,64 @@ def CreateNavigatonBar(root_path: str, max_levels: int, default_header_dict: dic
     list_of_strings = GenerateStrings(list_of_entries, is_locally_built)
     return list_of_strings
 
+def AddPythonSnippetOutputs(file_path: Path) -> None:
+    with open(file_path, "r") as file_input:
+        lines = file_input.readlines()
+
+    found_python_snippet_block = False
+    snippet_lines = []
+    output_lines = []
+    index = 0
+    while index < len(lines):
+        line = lines[index]
+        index += 1
+        if not found_python_snippet_block or line != "## POST_PROCESS_PAGES_PYTHON_OUTPUT_GENERATION\n":
+            output_lines.append(line)
+
+        if found_python_snippet_block and line.startswith("```"):
+            found_python_snippet_block = False
+            if "## POST_PROCESS_PAGES_PYTHON_OUTPUT_GENERATION\n" in snippet_lines:
+                # check whether existing expected output is found
+                temp_index = index
+                is_existing_output_found = False
+                while temp_index < len(lines):
+                    if lines[temp_index].strip() != "":
+                        is_existing_output_found  = lines[temp_index] == "Expected output:\n"
+                        is_existing_output_found &= lines[temp_index+1] == "```bash\n"
+                        break
+                    temp_index += 1
+
+                if not is_existing_output_found:
+                    # existing is not found or force re-write is enabled
+
+                    # create a temp file
+                    temp_file_path = f"{file_path.name}.temp.py"
+                    with open(temp_file_path, "w") as temp_file_output:
+                        temp_file_output.writelines(snippet_lines)
+
+                    subprocess_run = subprocess.run([GetPython3Command(), temp_file_path], stdout=subprocess.PIPE, universal_newlines=True)
```

Comment:
```suggestion
                    subprocess_run = subprocess.run([GetPython3Command(), temp_file_path], stdout=subprocess.PIPE, universal_newlines=True, check=True)
```

otherwise failures go undetected